File: 02-MvvmDataBinding/ViewModel/EmailViewModel.cs
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace MvvmDataBinding.ViewModel;

public class EmailViewModel : INotifyPropertyChanged
{
    private string? _subject;
    private string? _body;

    public EmailViewModel()
    {
    }

    public EmailViewModel(string subject, string body)
    {
        _subject = subject;
        _body = body;
    }

    public string? Subject
    {
        get => _subject;
        set
        {
            if (_subject == value)
            {
                return;
            }

            _subject = value;
            OnPropertyChanged();
        }
    }

    public string? Body
    {
        get => _body;
        set
        {
            if (_body == value)
            {
                return;
            }

            _body = value;
            OnPropertyChanged();
        }
    }

    public event PropertyChangedEventHandler? PropertyChanged;

    private void OnPropertyChanged([CallerMemberName] string? propertyName = null) =>
        PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));

    public override string ToString() => $"Subject: {Subject}\nBody: {Body}";
}

File: 02-MvvmDataBinding-VB/ViewModel/EmailViewModel.vb
Imports System.ComponentModel

Namespace Global.MvvmDataBinding.ViewModel

    Public Class EmailViewModel
        Implements INotifyPropertyChanged

        Private _subject As String
        Private _body As String

        Public Sub New()
        End Sub

        Public Sub New(subject As String, body As String)
            _subject = subject
            _body = body
        End Sub

        Public Property Subject As String
            Get
                Return _subject
            End Get
            Set(value As String)
                If _subject = value Then
                    Return
                End If

                _subject = value
                OnPropertyChanged(NameOf(Subject))
            End Set
        End Property

        Public Property Body As String
            Get
                Return _body
            End Get
            Set(value As String)
                If _body = value Then
                    Return
                End If

                _body = value
                OnPropertyChanged(NameOf(Body))
            End Set
        End Property

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        Protected Overridable Sub OnPropertyChanged(propertyName As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        End Sub

        Public Overrides Function ToString() As String
            Return $"Subject: {Subject}{vbLf}Body: {Body}"
        End Function

    End Class

End Namespace
